docs/schedule_scripting_language.md

```diff
@@ -0,0 +1,901 @@
+        - [`if_since_time` notes](#if_since_time-notes)
+    - [Unconditional branches](#unconditional-branches)
+      - [`branch`](#branch)
+        - [`branch` arguments](#branch-arguments)
+        - [`branch` example](#branch-example)
+        - [`branch` non-generics](#branch-non-generics)
+    - [`else`](#else)
+      - [`else` example](#else-example)
+    - [Return commands](#return-commands)
+      - [`return_s`](#return_s)
+        - [`return_s` arguments](#return_s-arguments)
+        - [`return_s` example](#return_s-example)
+      - [`return_l`](#return_l)
+        - [`return_l` arguments](#return_l-arguments)
+        - [`return_l` example](#return_l-example)
+      - [`return_none`](#return_none)
+        - [`return_none` arguments](#return_none-arguments)
+        - [`return_none` example](#return_none-example)
+      - [`return_empty`](#return_empty)
+        - [`return_empty` arguments](#return_empty-arguments)
+        - [`return_empty` example](#return_empty-example)
+      - [`return_time`](#return_time)
+        - [`return_time` arguments](#return_time-arguments)
+        - [`return_time` example](#return_time-example)
+    - [Miscellaneous commands](#miscellaneous-commands)
+      - [`nop`](#nop)
+        - [`nop` arguments](#nop-arguments)
+        - [`nop` example](#nop-example)
+    - [Operators](#operators)
+      - [`not`](#not)
+        - [`not` example](#not-example)
+    - [Labels](#labels)
+  - [Formal grammar](#formal-grammar)
+    - [Tokens](#tokens)
+
+## Features of the schedule system
+
+The Schedule system is a very simple scripting language, it is composed of a
+series of commands that can check the state of the game and return a value
+depending on said state. This system can't declare variables, hold its own
+state or modify the game's state.
+
+A schedule script can check the state of the following:
+
+- Current in-game [day](#if_day).
+- If a [WeekEventReg flag](#if_week_event_reg) is set or not.
+- Current time is in a specified [time range](#if_time_range).
+- Current time is [before](#if_before_time)/[after](#if_since_time) a specified
+  time.
+- Some other [miscellaneous](#if_misc) checks (see the `ScheduleCheckMisc`
+  enum).
+
+All of those checks act as conditional branches into some other command. The
+schedule system also supports unconditional branches.
+
+The schedule script is run until a return command is executed. There are various
+return commands that allow different things:
+
+- [Return none](#return_none): The schedule finished without returning a value.
+- [Return empty](#return_empty): The schedule finished without changing the
+  previous value.
+- [Return s](#return_s): The script returns a value that's 1 byte long.
+- [Return l](#return_l): The script returns a value that's 2 bytes long (Note
+  this is bugged and will be truncated to 1 byte).
+- [Return time](#return_time): Returns a time range and a 1 byte value.
```

Comment:
Is this 1 extra byte actor specific?

Reply:
Yes, it serves the same purpose as the returned value of `return_s`/`return_l`, but adding a time range.

Reply:
For your second question, the compiler is _very_ dumb, it doesn't even try to check for arguments being correct. It just assumes everything between parentheses is correct and passes them to the generated macros.
I made it this way to not have to worry to support about any symbol/macro/enum defined in C, just past it through and let the C compiler to handle them